artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/OperationContextImpl.java

```diff
@@ -304,6 +339,22 @@ public String toString() {
       }
    }
 
+   static final class StoreTaskHolder {
```

Comment:
Rename this to StoreOnlyTaskHolder

and add a comment please stating why this is separate from TaskHolder? (just a single line is fine, nothing big)

Reply:
Also add a comment on TaskHolder staying that we are not extending StoreTaskHolder because of memory optimizations.. as it would use more memory and this is on the hot path